src/foam/nanos/auth/AuthenticatedUserDAO.java

```diff
@@ -67,6 +72,21 @@ public FObject put_(X x, FObject obj) {
       toPut.setSpid(user.getSpid());
     }
 
+    if ( toPut != null ) {
+      User result = (User) super.find_(x, toPut.getId());
+      if (
+        result != null &&
+        ! SafetyUtil.equals(result.getGroup(), toPut.getGroup())
+      ) {
+        if ( updatingSelf ) {
+          throw new RuntimeException("You cannot set your own group.");
+        } else if ( ! isGlobalAdmin ) {
+          throw new RuntimeException("Only administrators may update another user's group.");
```

Comment:
I don't think this message or code is entirely accurate. I would just say that you aren't authorized to update the user's group. Also, I don't think we should check if you're a global admin, just if you have access to this group and user. We want to have weaker admins, like a GS Admin who can administer all of the groups and users under their control.